lib/poseidon/broker_pool.rb

```diff
@@ -65,8 +65,15 @@ def close
     private
     def fetch_metadata_from_broker(broker, topics)
       host, port = broker.split(":")
-      Connection.open(host, port, @client_id, @socket_timeout_ms, @connect_timeout_ms) do |connection|
+      port = port.to_i
```

Comment:
When we build connections from cluster metadata pulled from Kafka, the port will be an int.  If we leave this as a string, we'll never end up matching any connections on the next line.